problems/string-rotation/string-rotation.rs

```diff
@@ -0,0 +1,31 @@
+fn is_rotation(forward: &str, reverse: &str) -> bool {
```

Comment:
Check out this Python solution, which was quite insightful to me: https://github.com/mre/the-coding-interview/blob/master/problems/string-rotation/string-rotation.py#L21-L22
Maybe we can have the same in Rust?

Reply:
Actually, the magic bit in the Python solution is this: `and candidate in original*2`.
This is doing a substring lookup. The idea is, that the original string must appear *somewhere* when you double the string:

Example:
Original string: `hello`
Rotation: `ohell`
2*original: `hellohello`
As you can see, the rotation shows up in "2*original".